### ISSUE TITLE
reimplementation issues

### ISSUE THREAD

**Issue by huheng123** [akhilpm/DroneDetectron2#2]
Hello, when I use the check point and datasets in this project and execute the verification script, only 27.546AP, 48.8AP50, there is a certain gap with the paper, how to solve this problem? thanks
![image](https://user-images.githubusercontent.com/68471393/235554362-df00fdcf-9775-4a53-8280-ea1ca729c2fb.png)

Some categories have no value
![image](https://user-images.githubusercontent.com/68471393/235554538-bf01c24c-6de4-4aa4-82f9-cb869e7dbfc1.png)

**Comment by akhilpm** [comment]
This is due to a class-index mismatch in your implementation. Probably you are using a different dataloader function than mine. In my case, I haven't used the "ignored regions" class. See my dataloading helper function here:

https://github.com/akhilpm/DroneDetectron2/blob/main/croptrain/data/datasets/visdrone.py#L233

If you want to use the "ignored region" class comment that line and train your model. Also don't forget to change the id of the cluster/crop class when you do.

**Comment by huheng123** [comment]
How to view the clustering subgraph in the training process?
thank you

**Comment by akhilpm** [comment]
I don't understand what you mean "subgraph" in this context. If you want to visualize the clusters, you can customize the plot function in the utils:

https://github.com/akhilpm/DroneDetectron2/blob/main/utils/plot_utils.py

I used the following code to get slightly better plots for the paper.

```python
from detectron2.structures.instances import Instances
from matplotlib.patches import Rectangle
from PIL import Image
import matplotlib.pyplot as plt

def plot_detection_boxes(predictions, cluster_boxes, data_dict):
    img = Image.open(data_dict["file_name"])
    #crop_area = data_dict['crop_area']
    plt.axis('off')
    #x1, y1, x2, y2 = crop_area[0], crop_area[1], crop_area[2], crop_area[3]
    #img = img.crop((x1, y1, x2, y2))
    plt.imshow(img)
    ax = plt.gca()    
    if len(predictions)!=0:
        predictions = predictions[predictions.scores>0.6]
        predictions = predictions.pred_boxes.tensor.cpu()
        for bbox in predictions:
            x1, y1 = bbox[0], bbox[1]
            h, w = bbox[3]-bbox[1], bbox[2]-bbox[0]
            rect = Rectangle((x1, y1), w, h, linewidth=2, edgecolor='orange', facecolor='none')
            ax.add_patch(rect)
    if len(cluster_boxes)!=0:
        if isinstance(cluster_boxes, Instances):
            cluster_boxes = cluster_boxes.pred_boxes.tensor.cpu()
        for bbox in cluster_boxes:
            x1, y1 = bbox[0], bbox[1]
            h, w = bbox[3]-bbox[1], bbox[2]-bbox[0]
            rect = Rectangle((x1, y1), w, h, linewidth=2, edgecolor='r', facecolor='none')
            ax.add_patch(rect)
    plt.show()
    plt.clf()

```

**Comment by huheng123** [comment]
Thank you



发自我的iPhone


------------------ Original ------------------
From: Akhil P M ***@***.***&gt;
Date: Sat,May 6,2023 7:16 AM
To: akhilpm/DroneDetectron2 ***@***.***&gt;
Cc: shixiaominga ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [akhilpm/DroneDetectron2] reimplementation issues (Issue #2)





 
I don't understand what you mean "subgraph" in this context. If you want to visualize the clusters, you can customize the plot function in the utils:
 
https://github.com/akhilpm/DroneDetectron2/blob/main/utils/plot_utils.py
 
I used the following code to get slightly better plots for the paper.
 from detectron2.structures.instances import Instances from matplotlib.patches import Rectangle from PIL import Image import matplotlib.pyplot as plt def plot_detection_boxes(predictions, cluster_boxes, data_dict):     img = Image.open(data_dict["file_name"])     #crop_area = data_dict['crop_area']     plt.axis('off')     #x1, y1, x2, y2 = crop_area[0], crop_area[1], crop_area[2], crop_area[3]     #img = img.crop((x1, y1, x2, y2))     plt.imshow(img)     ax = plt.gca()         if len(predictions)!=0:         predictions = predictions[predictions.scores&gt;0.6]         predictions = predictions.pred_boxes.tensor.cpu()         for bbox in predictions:             x1, y1 = bbox[0], bbox[1]             h, w = bbox[3]-bbox[1], bbox[2]-bbox[0]             rect = Rectangle((x1, y1), w, h, linewidth=2, edgecolor='orange', facecolor='none')             ax.add_patch(rect)     if len(cluster_boxes)!=0:         if isinstance(cluster_boxes, Instances):             cluster_boxes = cluster_boxes.pred_boxes.tensor.cpu()         for bbox in cluster_boxes:             x1, y1 = bbox[0], bbox[1]             h, w = bbox[3]-bbox[1], bbox[2]-bbox[0]             rect = Rectangle((x1, y1), w, h, linewidth=2, edgecolor='r', facecolor='none')             ax.add_patch(rect)     plt.show()     plt.clf()
